sklearn/neighbors/base.py

```diff
@@ -581,7 +768,8 @@ def _radius_neighbors_reduce_func(self, dist, start,
             results = neigh_ind
         return results
 
-    def radius_neighbors(self, X=None, radius=None, return_distance=True):
+    def radius_neighbors(self, X=None, radius=None, return_distance=True,
+                         sort_results=False):
```

Comment:
Is sort_results=False backwards compatible??

Reply:
Why do we now need to provide `sort_results` control? Does it need to be lumped in with this PR?

Reply:
`sort_results=False` is the default in binary trees (see below), so `radius_neighbors` is backward compatible. 

https://github.com/scikit-learn/scikit-learn/blob/90a2c57951a9ae32e80246b171e4b47a74252090/sklearn/neighbors/binary_tree.pxi#L1351-L1352

However, `radius_neighbors_graph` now calls `radius_neighbors(sort_results=True)`, which makes the resulting sparse graph sorted by distance. This was not the case before, but I don't think we had any contract on the sorting order of the graph.
Now, this property is tested in `sklearn.neighbors.tests.test_graph.test_transformer_result`.

Reply:
With this API, the sorting is done in the binary tree, with Cython code. I expect it to be faster than in NumPy, it uses no GIL, and can be run in parallel threads since #10887.

https://github.com/scikit-learn/scikit-learn/blob/90a2c57951a9ae32e80246b171e4b47a74252090/sklearn/neighbors/binary_tree.pxi#L679-L689

We might want to add the same parameter in `radius_neighbors_graph`, defaulting to False, and calling with True only in the transformers.